pytorch_lightning/utilities/device_parser.py

```diff
@@ -100,7 +101,8 @@ def parse_tpu_cores(tpu_cores: Union[int, str, List]) -> Optional[Union[List[int
     if not _tpu_cores_valid(tpu_cores):
         raise MisconfigurationException("`tpu_cores` can only be 1, 8 or [<1-8>]")
 
-    if tpu_cores is not None and not _TPU_AVAILABLE:
+    # PL_TPU_AVAILABLE is used to mock TPU env.
+    if tpu_cores is not None and (not _TPU_AVAILABLE and os.getenv("PL_TPU_AVAILABLE", "0") != '1'):
```

Comment:
If you want this, it should be part of `_TPU_AVAILABLE`, not here.

And I'd argue this could go into a separate PR

Reply:
I tried to mock it but didn't manage to mock `_TPU_AVAILABLE` to `True`. It is need for a test.

Reply:
My point is, if we want this override, it should be

`_TPU_AVAILABLE = ... and os.getenv("PL_TPU_AVAILABLE", "0") != '1'`

where it is defined

Reply:
Done !